src/ai/backend/manager/models/scaling_group.py

```diff
@@ -216,13 +216,15 @@ async def query_allowed_sgroups(
     result = await db_conn.execute(query)
     from_domain = {row["scaling_group"] for row in result}
 
-    if isinstance(group, Iterable):
-        group_ids = await resolve_groups(db_conn, domain_name, group)
-    else:
-        if group_id := await resolve_group_name_or_id(db_conn, domain_name, group):
-            group_ids = [group_id]
-        else:
-            group_ids = []
+    match group:
+        case uuid.UUID() | str():
+            if group_id := await resolve_group_name_or_id(db_conn, domain_name, group):
+                group_ids = [group_id]
+            else:
+                group_ids = []
+        case list() | tuple() | set():
+            grps = await resolve_groups(db_conn, domain_name, cast(Iterable, group))
+            group_ids = [*grps]
```

Comment:
Is there a specific reason unpacking and repacking `list` instead of using as is?